packages/discovery-provider/src/tasks/create_engagement_notifications.py

```diff
@@ -0,0 +1,131 @@
+                UserChallenge.specifier == ChallengeDisbursement.specifier,
+            ),
+        )
+        .join(Challenge, Challenge.id == UserChallenge.challenge_id)
+        .outerjoin(
+            Notification,
+            and_(
+                Notification.specifier == UserChallenge.specifier,
+                Notification.group_id
+                == func.concat(
+                    CLAIMABLE_REWARD,
+                    ":",
+                    func.cast(UserChallenge.user_id, String),
+                    ":challenge:",
+                    func.cast(UserChallenge.challenge_id, String),
+                    ":specifier:",
+                    func.cast(UserChallenge.specifier, String),
+                ),
+            ),
+        )
+        .filter(
+            UserChallenge.is_complete,
+            UserChallenge.completed_at >= START_DATETIME,
+            UserChallenge.completed_at < datetime.now() - week_cooldown_interval,
+            Challenge.cooldown_days == 7,
+            ChallengeDisbursement.specifier.is_(None),  # no disbursement yet
+            Notification.id.is_(None),  # no notification yet
+        )
+        .limit(BATCH_SIZE)
+        .all()
+    )
+    for user_challenge in user_challenges_cooldown_completed:
+        time_threshold = user_challenge.completed_at - timedelta(hours=1)
+        # Query to check if there's already a notification within the last hour for this user
+        existing_notification = (
+            session.query(Notification)
+            .filter(
+                Notification.user_ids.any(
+                    user_challenge.user_id
+                ),  # Assumes 'user_ids' can handle multiple IDs and is searchable this way
+                Notification.timestamp >= time_threshold,
+                Notification.type == CLAIMABLE_REWARD,
+            )
+            .first()
+        )
+        if not existing_notification:
+            new_notification = Notification(
+                specifier=str(user_challenge.specifier),
+                group_id=get_claimable_reward_notification_group_id(
+                    user_challenge.user_id,
+                    user_challenge.challenge_id,
+                    user_challenge.specifier,
+                ),
+                blocknumber=user_challenge.completed_blocknumber,
+                user_ids=[user_challenge.user_id],
+                type=CLAIMABLE_REWARD,
+                data={
+                    "specifier": user_challenge.specifier,
+                    "challenge_id": user_challenge.challenge_id,
+                    "amount": user_challenge.amount,
+                },
+                timestamp=user_challenge.completed_at,
+            )
+            new_notifications.append(new_notification)
+        else:
+            logger.info(f"Debouncing notification for {user_challenge.user_id}")
+    logger.info(f"Inserting {len(new_notifications)} claimble reward notifications")
```

Comment:
prob fine, but do we want to not log if new_notifications length is 0?

Reply:
this only logs once every 10 mins and maybe useful to know if it's missing any 🤷